src/main/java/org/folio/entlinks/service/links/LinksSuggestionService.java

```diff
@@ -177,16 +176,35 @@ private void actualizeBibSubfields(FieldParsedContent bibField,
       return;
     }
 
-    var zeroValue = sourceFileCodeRepository.findByCodeAsPrefixFor(authority.getNaturalId())
-        .map(AuthoritySourceFileCode::getAuthoritySourceFile)
-        .map(sourceFile -> getSubfield0Value(authority.getNaturalId(), sourceFile))
-        .orElse(authority.getNaturalId());
+    var zeroValue = getSubfieldZeroValue(authority);
+
     bibSubfields.put("0", List.of(zeroValue));
     bibSubfields.put("9", List.of(authority.getId().toString()));
 
     modifySubfields(optionalField.get().getSubfields(), bibSubfields, rule);
   }
 
+  private String getSubfieldZeroValue(AuthorityParsedContent authority) {
+    var sourceFileId = authority.getSourceFileId();
+    if (isNull(sourceFileId)) {
+      return authority.getNaturalId();
+    }
+
+    var authoritySourceFile = authoritySourceFileRepository.findById(sourceFileId);
+    var naturalIdPrefix = extractNaturalIdPrefix(authority.getNaturalId());
+
+    return authoritySourceFile.flatMap(file ->
+        file.getAuthoritySourceFileCodes().stream()
+            .filter(code -> code.getCode().equals(naturalIdPrefix))
+            .findFirst()
+            .map(code -> getSubfield0Value(authority.getNaturalId(), file))
+    ).orElse(authority.getNaturalId());
+  }
+
+  private String extractNaturalIdPrefix(String naturalId) {
+    return naturalId.split("[0-9]")[0];
```

Comment:
if natural id does not start with letters then here I think you would get index out of bounds error. 
I propose first to check if the prefix is alphabetic and if yes then to extract and return it.